Objects/setobject.c

```diff
@@ -1185,7 +1202,9 @@ make_new_frozenset(PyTypeObject *type, PyObject *iterable)
         /* frozenset(f) is idempotent */
         return Py_NewRef(iterable);
     }
-    return make_new_set(type, iterable);
+    PyObject *obj = make_new_set(type, iterable);
+    _PyFrozenSet_MaybeUntrack(obj);
+    return obj;
```

Comment:
I would prefer to check for NULL here. Same suggestion for PyFrozenSet_New().

```suggestion
    PyObject *obj = make_new_set(type, iterable);
    if (obj != NULL) {
        _PyFrozenSet_MaybeUntrack(obj);
    }
    return obj;
```

Reply:
That is fine. I will then replace the NULL check inside `_PyFrozenSet_MaybeUntrack` with an assert.